v23.1/architecture/transaction-layer.md

```diff
@@ -50,11 +50,11 @@ CockroachDB provides the following types of reads:
 
 ### Commits (phase 2)
 
-CockroachDB checks the running transaction's record to see if it's been `ABORTED`; if it has, it restarts the transaction.
+CockroachDB checks the running transaction's record to see if it's been `ABORTED`; if it has, it [throws a retryable error](../transaction-retry-error-reference.html) to the client.
 
 In the common case, it sets the transaction record's state to `STAGING`, and checks the transaction's pending write intents to see if they have succeeded (i.e., been replicated across the cluster).
```

Comment:
```suggestion
In the common case, it sets the transaction record's state to `STAGING` and checks the transaction's pending write intents to see if they have been successfully replicated across the cluster.
```
I'm confused about what "In the common case" refers to. I'm also confused about how line 53 relates to line 55 and 57. Who would have set the transaction record to ABORTED? If it's ABORTED, we throw the retryable error and move on. Otherwise, WE set the transaction record's state to STAGING (what else might it have been before?) and check for successful write attempts. When those succeed, then what is the status of the transaction record's state?

Reply:
This part I did not work on specifically other than add a link, however, I do think you're right that the second paragraph is weirdly phrased. I am adjusting it to clear up some of those odd areas